src/java.base/share/classes/java/nio/file/Path.java

```diff
@@ -248,6 +249,62 @@ public static Path of(URI uri) {
      */
     Path getFileName();
 
+    /**
+     * Returns the file name extension of this path as a {@code String}. The
+     * extension is the portion of the {@linkplain #getFileName file name}
+     * string after the last period character ('.', U+002E FULL STOP). Leading
+     * period characters in the string are ignored. If there is no  extension,
+     * then {@code null} is returned. This will occur if the path has zero
+     * elements ({@link #getFileName()} returns {@code null}), the file name
+     * string does not contain a period character, or only the first character
+     * is a period character. If the last character is a period character but
+     * some other character is not, then the extension is the
+     * {@linkplain String#isEmpty() empty} string.
```

Comment:
> ... or only the first character is a period character.

This needs to be updated to be consistent with the "all leading periods" rule, something like

> ... or all the period characters are leading periods.

Also:

> If the last character is a period character but some other character is not, then the extension is the
> empty string.

I'm not sure what this is trying to say. Is it saying, if the last character is a period, and it is not a leading period, then the extension is empty? If so, then maybe it could say just that; or possibly the statement could be removed since it's a consequence of the definition given earlier.